### PR TITLE
Fix file:// syncs  deleting the original files

### LINKED ISSUE #4681: file:// sync deletes files from directory
**Version**
Pulpcore 3.39

**Describe the bug**
When syncing file:// repositories, files are disappearing after the sync.

**To Reproduce**
1) Copy these two repositories to the FS:
  - https://github.com/Katello/katello/tree/master/test/fixtures/test_repos/file1
  - https://github.com/Katello/katello/tree/master/test/fixtures/test_repos/file2
2) Sync one, then the other
3) See that some files disappeared.
  - In my case, file2 lost every file except PULP_MANIFEST

**Expected behavior**
No files disappear.

**Additional context**
This also occurred with RPM content type files.

### PR DESCRIPTION
**Pull request by dralley**
closes #4681

### CODE REVIEW

File: pulpcore/tests/functional/api/pulp_file/test_sync.py [hunk 1]
```diff
@@ -22,17 +24,20 @@ def test_sync_file_protocol_handler(
...
     }
     remote = gen_object_with_cleanup(file_remote_api_client, remote_kwargs)
+    files = set(os.listdir("/tmp/file/"))
 
     body = RepositorySyncURL(remote=remote.pulp_href)
     monitor_task(file_repository_api_client.sync(file_repo.pulp_href, body).task)
 
+    # test that all the files are still present
+    assert set(os.listdir("/tmp/file/")) == files
+
```
Comment (mdellweg) [inline pulpcore/tests/functional/api/pulp_file/test_sync.py:40]: +1

**Comment by dralley** [comment]
I noticed that we're copying and resaving the file in the temporary directory even in the file:// sync case.  That's not especially efficient and we should file an issue for it, but in the meantime we can simply make sure the sync process is working with that copy rather than the original.

**Comment by mdellweg** [comment]
> I noticed that we're copying and resaving the file in the temporary directory even in the file:// sync case. That's not especially efficient and we should file an issue for it, but in the meantime we can simply make sure the sync process is working with that copy rather than the original.

Is that the `path` vs. `_path` change?

**Comment by mdellweg** [comment]
Should we then bring the reverted change to delete downloaded files early back? Or was the revert never merged?

**Comment by dralley** [comment]
@mdellweg It was never reverted, this is the whole fix.  `self._path` -> `self.path`, that is.

**Review by mdellweg — APPROVED** [review]

**Review by ipanova — APPROVED** [review]
can you squash into one commit?

**Comment by mdellweg** [comment]
I believe it's on purpose, because the test itself is not backportable? Does that sound right @dralley ?

**Pull request merged**